packages/bridge/src/vl/utils/conversion/category/index.ts

```diff
@@ -10,11 +10,16 @@ import { findColorForCategory } from '../histogram';
  * @returns {object[]}
  */
 export function vlToCategory(histogram: VLCategoricalHistogram, legendData: any = []): object[] {
-  return histogram.value.map((d) => ({
-    color: findColorForCategory(d.x, legendData),
-    name: d.x,
-    value: d.y
-  }));
+  const fullCategories = (histogram as any)._categories;
```

Comment:
Hmm, we're using a private property here. We need to be careful as it can change suddenly without no one noticing that is being used here. I guess that having a public property here is difficult but that's what we need to achieve.

Moreover, I think we should avoid using `any` at all times. If we're gonna use `_categories` here we should add it to the current type.

Reply:
To get this done we would need to type everything from histogram to the top, because right now `_dataLayer`, `carto`, `layer`, the widgets,... all of them are set to `any`. I could add `_categories` to `VLCategoricalHistogram` interface, but it would be just a patch that would hide the real problem.
IMO, it would be great to do it, but it deserves its own task to focus and set the right types everywhere it's needed, and not just here; maybe blocking this feature until then if you think that's needed.

Reply:
Yes, I mean, it would be great to have everything with the proper typing but we'd have needed to start since the very beginning.

What I meant is that `(histogram as any)` cast is not good, and even if `_categories` is a private property we should add it to the typing if we're going to use it.

Reply:
Ok, not I understand. I put this cast to emphasize that private variable access; but I think you are right, maybe it'd be better to set it in the interface with a comment and create a task to make it  "official" until it is broken in the future.

Reply:
I agree with @jesusbotella. We should find a better way to access private categories in this case. What about adding a getter to the VLCategoricalHistogram class "getAllCategories"?

Reply:
It is an interface, I rather type what's coming from this (`dataLayer.viz.variables`) from its library, and set `categories` as a public property or add a public method to get this categories.

Reply:
@jesusbotella @VictorVelarde what do you think about this? It's OK for you (for now) to use `_categories` directly, or with a method. With that decision, we can unblock the PR and merge it

Reply:
I guess we all prefer a public method, to officialize that it is accessed from outside. I don't have a strong opinion its the naming, so as you wish